eventuals/type-traits.h

```diff
@@ -172,6 +173,32 @@ using tuple_types_union_t = typename tuple_types_union<Left, Right>::type;
 
 ////////////////////////////////////////////////////////////////////////
 
+template <typename...>
+struct tuple_types_subtract {
+  using type = std::tuple<>;
+};
+
+template <typename Left, typename... Lefts, typename Right>
+struct tuple_types_subtract<
+    std::tuple<Left, Lefts...>,
+    Right> {
+  using type = tuple_types_concatenate_t<
+      std::conditional_t<
+          std::disjunction_v<
+              tuple_types_contains<std::exception, Right>,
+              tuple_types_contains<std::exception_ptr, Right>,
```

Comment:
You're writing a _generic_ function called `tuple_types_subtract`, what does that have to do with `std::exception` and `std::exception_ptr`!?